### PR TITLE
Data Layer: Cleanup `retryOnFailure` tests

### PR DESCRIPTION
**Pull request by tyxla**
#### Changes proposed in this Pull Request

This PR is a follow-up cleanup to https://github.com/Automattic/wp-calypso/pull/63816#pullrequestreview-978621278

#### Testing instructions

Verify tests pass: `yarn run test-client client/state/data-layer/wpcom-http/pipeline/retry-on-failure/test/index.js`

### CODE REVIEW

File: client/state/data-layer/wpcom-http/pipeline/retry-on-failure/test/index.js [hunk 1]
```diff
@@ -24,128 +26,121 @@ const withRetries = ( retryCount ) => ( actionOrInbound ) =>
...
+		expect( inbound.store.dispatch.mock.calls.length ).toEqual( 2 );
+		expect( inbound.store.dispatch ).toBeCalledWith( withRetries( 2 )( originalRequest ) );
 
 		// retry 3
 		expect(
-			retryIt( { ...inbound, originalRequest: dispatch.mock.lastCall[ 0 ] } )
+			retryIt( { ...inbound, originalRequest: inbound.store.dispatch.mock.lastCall[ 0 ] } )
 		).toHaveProperty( 'shouldAbort', true );
-		expect( dispatch.mock.calls.length ).toEqual( 2 );
+		expect( inbound.store.dispatch.mock.calls.length ).toEqual( 2 );
```
Comment (jsnajdr) [inline client/state/data-layer/wpcom-http/pipeline/retry-on-failure/test/index.js]: Unless I miss some subtle detail, `mock.calls.length` is exactly the same as `.toBeCalledTimes( x )`, would be nice to use the dedicated method.

Looking directly at `mock.calls` should be needed only when checking for some special condition.

**Review by jsnajdr — APPROVED** [review]
:+1:

File: client/state/data-layer/wpcom-http/pipeline/retry-on-failure/test/index.js (hunk 1 above)
Comment (tyxla) [reply client/state/data-layer/wpcom-http/pipeline/retry-on-failure/test/index.js]: Yep! You're not missing anything. It's just the consequence of speedy migration 😅

Comment (tyxla) [reply client/state/data-layer/wpcom-http/pipeline/retry-on-failure/test/index.js]: Addressed in https://github.com/Automattic/wp-calypso/pull/63862/commits/3ea450d170be6f4b1c1bdad30c2a17cb1ae47c18

**Pull request merged**